minerl/herobraine/hero/handlers/agent/observations/inventory.py

```diff
@@ -93,35 +96,29 @@ def from_hero(self, obs):
             assert stack["quantity"] >= 0
             assert stack["metadata"] in range(16)
             if key is not None:
+                if key not in self.items:
+                    key = self._other
                 item_dict[key] += stack["quantity"]
```

Comment:
```suggestion
            if key is not None:
                item_dict[key] += stack["quantity"]
            elif self._other in self.items:
               item_dict[self._other] += stack["quantity"]
```

Reply:
"other" isn't in any of the FlatInventoryObservation item lists for any of our EnvSpecs (include ObtainDiamond), show probably should check before inserting this into the returned item_dict.

Reply:
You mean check that this works? It will only be added if self._other is in self.items

Reply:
Ah, by "check" I meant add the `elif self._other` check.

Reply:
OK I adjusted test_inventory_obs.py to reflect the correct behavior